bin/reth/src/node/mod.rs

```diff
@@ -278,6 +278,17 @@ impl Command {
         debug!(target: "reth::cli", "Spawning payload builder service");
         ctx.task_executor.spawn_critical("payload builder service", payload_service);
 
+        // TODO(alexey): don't spawn pruning task if pruning is not enabled via config from https://github.com/paradigmxyz/reth/pull/3341
```

Comment:
https://github.com/paradigmxyz/reth/pull/3341 is good to merge and shouldn't interfere with anything I think